packages/odyssey-react-mui/src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,208 @@
+   */
+  value?: unknown;
+}
+
+export const TextField = forwardRef<HTMLInputElement, TextFieldProps>(
+  (props, ref) => {
+    const {
+      autoCompleteType,
+      isDisabled = false,
+      endAdornment,
+      errorMessage,
+      hintText,
+      id: idOverride,
+      labelText,
+      isMultiline = false,
+      onChange,
+      onFocus,
+      optionalLabelText,
+      placeholderText,
+      isReadOnly,
+      isRequired = true,
+      startAdornment,
+      type: inputType = "text",
+      value,
+    } = props;
+
+    const [inputType, setInputType] = useState(inputType);
+
+    const togglePasswordVisibility = useCallback(() => {
+      setInputType((currentType) =>
+        currentType === "password" ? "text" : "password"
+      );
+    }, []);
+
+    const id = useUniqueId(idOverride);
+    const hintId = hintText ? `${id}-hint` : undefined;
+    const errorId = errorMessage ? `${id}-error` : undefined;
+    const labelId = labelText ? `${id}-label` : undefined;
+
+    return (
+      <FormControl disabled={isDisabled} error={Boolean(errorMessage)}>
+        <InputLabel htmlFor={id} id={labelId}>
+          {labelText}
+          {!isRequired && (
+            <Typography variant="subtitle1">{optionalLabelText}</Typography>
+          )}
+        </InputLabel>
+        {hintText && <FormHelperText id={hintId}>{hintText}</FormHelperText>}
+        <InputBase
+          autoComplete={autoCompleteType}
+          endAdornment={
+            inputType === "password" ? (
+              <InputAdornment position="end">
+                <IconButton
+                  aria-label="toggle password visibility"
+                  edge="end"
+                  onClick={togglePasswordVisibility}
+                >
+                  {inputType === "password" ? <EyeIcon /> : <EyeOffIcon />}
+                </IconButton>
+              </InputAdornment>
+            ) : (
+              endAdornment
+            )
+          }
+          id={id}
+          inputProps={
+            hintId || errorId
+              ? {
+                  "aria-describedby":
+                    hintId && errorId
+                      ? `${hintId} ${errorId}`
+                      : hintId || errorId,
+                }
+              : undefined
+          }
+          multiline={isMultiline}
+          onChange={onChange}
+          onFocus={onFocus}
+          placeholder={placeholderText}
+          readOnly={isReadOnly}
+          ref={ref}
+          startAdornment={
+            inType === "search" ? (
+              <InputAdornment position="start">
+                <SearchIcon />
+              </InputAdornment>
+            ) : (
+              startAdornment
+            )
+          }
+          type={inputType}
+          value={value}
+        />
+        {errorMessage && (
+          <FormHelperText error id={errorId}>
+            <span style={visuallyHidden}>Error:</span>
```

Comment:
This needs to be internationalized.

Reply:
Internationalization is out of scope for this PR. It will be a significant project and coming in the future.